### PR TITLE
Add an automerge workflow for provider and cli docs

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #5298

This new automerge functionality will mean that provider docs PRs
+ CLI docs PRs will get a new tag added `automation/merge` that if
present AND the checks for the PR pass, then the PR will be automerged

This workflow will work for *any* PR that is tagged with
`automation/merge` - the nice point here is that a non-pulumiam
cannot add labels to PRs

### CODE REVIEW

File: .github/workflows/automerge-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+jobs:
+  automerge:
+    name: Automerge Workflow
+    runs-on: ubuntu-latest
+    steps:
+      - env:
+          GITHUB_TOKEN: ${{ secrets.PULUMI_BOT_TOKEN }}
+          MERGE_COMMIT_MESSAGE: pull-request-title
+          MERGE_FORKS: "false"
+          MERGE_LABELS: automation/merge
```
Comment (cnunciato) [inline .github/workflows/automerge-workflow.yml:10]: Just to check my understanding: It's the presence of this label on _any_ PR that will result in an auto-merge on successful completion of the check suite. Correct? And this label only gets applied (automatically) with `build-tfgen-provider-docs` and `build-pulumi-cli-docs`?

Comment (stack72) [reply .github/workflows/automerge-workflow.yml:10]: Correct - this means that the use case can be extended where required, but the only place that adds it by default is the provider and cli docs PRs

Comment (stack72) [reply .github/workflows/automerge-workflow.yml:10]: We can add any tags in here that you want, but it still allows people in our org to get around the restrictions - this is a process / cultural problem if it happens right?

Comment (cnunciato) [reply .github/workflows/automerge-workflow.yml:10]: Ok, thanks for confirming. cc @leezen just as a head's up. I'm okay to try it as long as no objections from others.

**Review by cnunciato — APPROVED** [review]

**Review by davidwrede — APPROVED** [review]

**Pull request merged**